examples/advanced_stream.rs

```diff
@@ -0,0 +1,46 @@
+use std::env;
+use std::time::Duration;
+
+use futures::stream::StreamExt;
+
+use deepgram::{
+    transcription::prerecorded::options::{Language, Options},
+    Deepgram, DeepgramError,
+};
+
+#[tokio::main]
+async fn main() -> Result<(), DeepgramError> {
+    let dg = Deepgram::new(env::var("DEEPGRAM_API_KEY").unwrap());
+
+    let options = Options::builder()
+        .smart_format(true)
+        .language(Language::en_US)
+        .build();
+
+    let mut results = dg
+        .transcription()
+        .stream_request_with_options(Some(&options))
+        .keep_alive()
+        .encoding("linear16".to_string())
+        .sample_rate(44100)
+        .channels(2)
+        .endpointing("300".to_string())
```

Comment:
- true
- false
- number

Reply:
Booleans are generally not very self-documenting, so I'd recommend something like:

```rust
pub enum Endpointing {
    Enabled,
    Disabled,
    CustomDuration(std::time::Duration),
}

impl serde::Serialize for Endpointing {
    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error> 
    where
        S: serde::Serializer,
    {
        match self {
            Endpointing::Enabled => serializer.serialize_str("enabled"),            
            Endpointing::Disabled => serializer.serialize_str("disabled"),
            Endpointing::CustomDuration(d) => serializer.serialize_u128(d.as_millis()),
        }
    }
```

(or if the number represents something else, just use `u64` instead of `Duration`.

Reply:
Was not able to get this working with an enum

Found some similar code here but it does not use an enum

https://github.com/deepgram-devs/deepgram-rust-sdk/blob/main/src/usage/get_usage_options.rs#L472